web/controllers/pair_retro_controller.ex

```diff
@@ -67,35 +70,42 @@ defmodule Pairmotron.PairRetroController do
   end
 
   def edit(conn = @authorized_conn, _params) do
-    conn = assign(conn, :projects, Repo.all(Project))
-    retro = conn.assigns.pair_retro
-    changeset = PairRetro.changeset(retro, %{}, nil)
-    render(conn, "edit.html", pair_retro: retro, changeset: changeset)
+    retro = conn.assigns.pair_retro |> Repo.preload(:pair)
+    projects = Project.projects_for_group(retro.pair.group_id) |> Repo.all
+    changeset = PairRetro.changeset(retro, %{}, nil, nil)
+    render(conn, "edit.html", pair_retro: retro, changeset: changeset, projects: projects)
   end
   def edit(conn, _params) do
     redirect_not_authorized(conn, pair_retro_path(conn, :index))
   end
 
   def update(conn = @authorized_conn, %{"pair_retro" => pair_retro_params}) do
-    pair_retro = conn.assigns.pair_retro |> Repo.preload(:pair)
+    pair_retro = conn.assigns.pair_retro |> Repo.preload([:pair, :project])
     pair = pair_retro.pair
 
-    earliest_pair_date = Pairmotron.Calendar.first_date_of_week(pair.year, pair.week)
-    changeset = PairRetro.update_changeset(pair_retro, pair_retro_params, earliest_pair_date)
+    project = project_from_params_or_pair_retro(pair_retro_params, pair_retro)
+
+    changeset = PairRetro.update_changeset(pair_retro, pair_retro_params, pair, project)
 
     case Repo.update(changeset) do
       {:ok, pair_retro} ->
         conn
         |> put_flash(:info, "Pair retro updated successfully.")
         |> redirect(to: pair_retro_path(conn, :show, pair_retro))
       {:error, changeset} ->
-        render(conn, "edit.html", pair_retro: pair_retro, changeset: changeset)
+        projects = Project.projects_for_group(pair.group_id) |> Repo.all
+        render(conn, "edit.html", pair_retro: pair_retro, changeset: changeset, projects: projects)
     end
   end
   def update(conn, _params) do
     redirect_not_authorized(conn, pair_retro_path(conn, :index))
   end
 
+  defp project_from_params_or_pair_retro(params, pair_retro) do
+    id = Map.get(params, "project_id") || (pair_retro.project && pair_retro.project.id) || 0
```

Comment:
The `|| 0` is here so that the `Repo.get` works right? I think it would be better to return `nil` instead of making a database call and relying on there never being a Project 0.